x-pack/filebeat/module/iptables/log/test/geo.log-expected.json

```diff
@@ -3,10 +3,15 @@
         "destination.ip": "10.4.0.5",
         "destination.mac": "90:10:20:76:8d:20",
         "destination.port": 443,
+        "event.category": "network",
         "event.dataset": "iptables.log",
+        "event.kind": "event",
         "event.module": "iptables",
-        "event.outcome": "deny",
```

Comment:
I think it still makes sense to populate this if we know what action was taken, but use `success` or `failure`.

Reply:
I'll ask for some ECS input.  My understanding was that allowed/denied would move to event.type.  So you could have:
event.type = ["connection", "denied"]

and then event.outcome could be success or failure, based on if you were successful in denying the connection.

Reply:
@leehinman that's correct.

Reply:
> and then event.outcome could be success or failure, based on if you were successful in denying the connection.

So isn't the fact that iptables says it blocked the connection evidence enough to set the `event.outcome` to `failure`?

Reply:
Yeah I'd have to agree with @andrewkroh here. When we model the iptables events as category "network" and type "connection", looking for `event.outcome:failure` I would expect to find the connections denied. Not events where a firewall "failed to block a connection" -- can this actually happen?

Reply:
If iptables failed to block the connection I'd be very surprised if we got something in the iptables log saying so.  At best we might get something from kernel logger.  So I'd expect every allowed/denied to end up with event.outcome = success.  Which doesn't make event.outcome very useful.

Reply:
out of band yes, its possible (websense & some IPS's used to work like that - it sees badness - sends a spoofed reset oob to kill the connection, and it didn't always work).

from an observers standpoint wouldn't the "deny" be event.action? with outcome being success/deny (with a strong default to succeed as oob is likely fairly limited in production these days). 

iptables is an ugly one as it can be e.g. ubiquiti as a network firewall, or a host based firewall - where the connection event (e.g. from auditbeat) would have a event.outcome as fail, but e.g. filebeat pulling in the iptables event would be event.action:deny | block | reset, etc. with the event.outcome being specific to the action being taken (in this case the block by the firewall component)

Reply:
sorry - "denied" can be one of the values in the event.type array, as well event.action: deny

Reply:
Yeah I'm retreating from my previous comment. I agree with setting `event.outcome:successful` when a firewall denies a connection.

Examples of `event.outcome:failure` would be cases like @dainperkins mentions, where for example sending a TCP rst fails to kill the connection. Whether or not a given firewall correctly reports this is not relevant in the bigger picture (we'd simply never set it in this case).

There's other well known cases where the agent's action is not successful, like Endpoint trying to kill a process, but the process is already gone.